repose-aggregator/components/filters/compression-filter/src/main/scala/org/openrepose/filters/compression/CompressionFilter.scala

```diff
@@ -91,6 +93,7 @@ class CompressionFilter @Inject()(configurationService: ConfigurationService, co
           logger.error("Error with the CompressingFilter", e)
           httpResponse.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR)
       }
+      httpResponse.commitToResponse()
```

Comment:
So the following discussion kind of details why this change is "needed":
https://sourceforge.net/p/pjl-comp-filter/discussion/378427/thread/31dd22c2/

The PJL Compressing Filter does not (and, in fact, cannot) set the Content-Length header since it processes the body as a stream.

Our wrapper, on the other hand, will buffer the response until committed, allowing us to set the content-length at the time of commit. That is, our wrapper is where the content-length is actually being set. As @wdschei pointed out, the PowerFilter necessarily performs this function since parts of the response must be known for RMS to function.

The Content-Length header is not strictly required by HTTP spec, but if it is not present, either transfer-encoding should be set to chunked, or the connection should be closed after the body has been written. Some if not all containers will handle this behavior for the user.

I'm not sure if we need a unit test to check the content-length header or not -- the underlying filter does not set it, that is additional (optional) processing that we perform in the wrapping filter. It seems to me that we could leave that to the PowerFilter and save one instance of breaking the stream.